packages/frontend-2/plugins/survicate.client.ts

```diff
@@ -0,0 +1,100 @@
+import { useOnAuthStateChange } from '~/lib/auth/composables/auth'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { useSynchronizedCookie } from '~/lib/common/composables/reactiveCookie'
+import dayjs from 'dayjs'
+import type { Survicate } from '@survicate/survicate-web-surveys-wrapper'
+import type { Nullable } from '@speckle/shared'
+
+export default defineNuxtPlugin(async () => {
+  const { isLoggedIn } = useActiveUser()
+  let survicateInstance = null as Nullable<Survicate>
+
+  if (!isLoggedIn.value) {
+    return {
+      provide: {
+        survicate: survicateInstance
+      }
+    }
+  }
+
+  const {
+    public: { survicateWorkspaceKey }
+  } = useRuntimeConfig()
+
+  const logger = useLogger()
+  const onAuthStateChange = useOnAuthStateChange()
+  const prepareSurvey = useInitMainSurvey()
+
+  try {
+    const { initSurvicate, getSurvicateInstance } = await import(
+      '@survicate/survicate-web-surveys-wrapper/widget_wrapper'
+    )
+    await initSurvicate({ workspaceKey: survicateWorkspaceKey })
```

Comment:
We should probably skip all this if `!survicateWorkspaceKey?.length`, I dunno what's gonna happen if u try to initialize it w/ an empty key